src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -218,26 +218,31 @@ static bool ShouldGenerateSourceGenInitializeComponent(XamlProjectItemForIC xaml
 
 		var rootClass = root.Attributes["Class", XamlParser.X2006Uri]
 					 ?? root.Attributes["Class", XamlParser.X2009Uri];
+		INamedTypeSymbol? rootType;
 
-		if (rootClass == null)
-			return false;
-
-		XmlnsHelper.ParseXmlns(rootClass.Value, out var rootTypeName, out var rootClrNamespace, out _, out _);
+		if (rootClass != null)
+		{
+			XmlnsHelper.ParseXmlns(rootClass.Value, out var rootTypeName, out var rootClrNamespace, out _, out _);
+			rootType = compilation.GetTypeByMetadataName($"{rootClrNamespace}.{rootTypeName}");
+		}
+		else
+		{ //no x:Class, but it can be an autogenerated type (starting with __Type, and with a XamlResourceId attribute)
+			ITypeSymbol xamlResIdAttr = compilation.GetTypeByMetadataName("Microsoft.Maui.Controls.Xaml.XamlResourceIdAttribute")!;
+			INamedTypeSymbol? GetTypeForResourcePath(string resourcePath, IAssemblySymbol assembly)
+			{
+				var attr = assembly.GetAttributes(xamlResIdAttr).FirstOrDefault(attr => (string)attr.ConstructorArguments[1].Value! == resourcePath);
```

Comment:
Looking at the line below, I think you can use `?` instead of `!`:
```suggestion
				var attr = assembly.GetAttributes(xamlResIdAttr).FirstOrDefault(attr => (string)attr?.ConstructorArguments[1].Value == resourcePath);
```

Reply:
resourcePath is non-nullable, so we can't compare. also, all XRID are generated, so if a actor arg is null, we're in trouble anyway